pkg/targetallocator/adapters/config_to_prom_config.go

```diff
@@ -59,6 +78,201 @@ func ConfigToPromConfig(cfg string) (map[interface{}]interface{}, error) {
 	return prometheus, nil
 }
 
+// UnescapeDollarSignsInPromConfig replaces "$$" with "$" in the "replacement" fields of
+// both "relabel_configs" and "metric_relabel_configs" in a Prometheus configuration file.
+func UnescapeDollarSignsInPromConfig(cfg string) (map[interface{}]interface{}, error) {
+	prometheus, err := ConfigToPromConfig(cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	prometheusConfigProperty, ok := prometheus["config"]
+	if !ok {
+		return nil, errorNoComponent("prometheusConfig")
+	}
+
+	prometheusConfig, ok := prometheusConfigProperty.(map[interface{}]interface{})
+	if !ok {
+		return nil, errorNotAMap("prometheusConfig")
+	}
+
+	scrapeConfigsProperty, ok := prometheusConfig["scrape_configs"]
+	if !ok {
+		return nil, errorNoComponent("scrape_configs")
+	}
+
+	scrapeConfigs, ok := scrapeConfigsProperty.([]interface{})
+	if !ok {
+		return nil, errorNotAList("scrape_configs")
+	}
```

Comment:
I don't think we should error if these don't exist, but rather just exit. Target allocator doesn't require Prometheus configuration.

Maybe this should be done in a separate PR though, as I believe this mirrors the current behaviour. WDYT @jaronoff97 ?

Reply:
@jaronoff97 @swiatekm-sumo I can create a follow up PR to relax this restriction.